AzureStorageBlob/Source/Models/BlobProperties.swift

```diff
@@ -151,6 +151,48 @@ public final class BlobProperties: XMLModelProtocol {
             "RemainingRetentionDays": XMLMetadata(jsonName: "remainingRetentionDays"),
         ])
     }
+
+    internal init(from headers: HttpHeaders) {
+        self.creationTime = headers["x-ms-creation-time"].asDate
```

Comment:
Should we convert all of these hardcoded strings to cases of `HttpHeader`? We'd need to add some cases but I feel like that's a best practice. If there are `x-` headers custom to Storage, we could expose those in a `StorageHttpHeader` enum.

Reply:
I think that's a good idea. Unfortunately, you can't extend enums in Swift to add cases. 

What we can do is have each library declare a Headers file with its unique headers, and give it the same extensions as the AzureCore one. Common headers would bubble up to AzureCore.